pkg/gencosmosdb/cosmosdb/cosmosdb.go

```diff
@@ -72,16 +73,30 @@ func (c *databaseClient) authorizeRequest(req *http.Request, resourceType, resou
 }
 
 func (c *databaseClient) do(ctx context.Context, method, path, resourceType, resourceLink string, expectedStatusCode int, in, out interface{}, headers http.Header) error {
+	for {
```

Comment:
Later we will need a metric on this.
Perhaps we should add a log line now? (That might involve adding logrus as a dependency but that's OK with me)
We also need a configurable limit on the number of retries.

Reply:
follow-up issue to add a metric

Reply:
follow-up issue for some sort of load/exponential backoff test

Reply:
https://github.com/Azure/ARO-RP/issues/92 - Metrics

Reply:
https://github.com/Azure/ARO-RP/issues/94 - load test